apps/papra-client/src/modules/documents/pages/document.page.tsx

```diff
@@ -85,6 +90,43 @@ export const DocumentPage: Component = () => {
 
   const getDataUrl = () => queries[1].data ? URL.createObjectURL(queries[1].data) : undefined;
 
+  const [isEditing, setIsEditing] = createSignal(false);
+  const [editedContent, setEditedContent] = createSignal('');
+  const [isSaving, setIsSaving] = createSignal(false);
+
+  const handleEdit = () => {
+    setEditedContent(queries[0].data?.document.content ?? '');
+    setIsEditing(true);
+  };
+
+  const handleCancel = () => {
+    setIsEditing(false);
+    setEditedContent('');
+  };
+
+  const handleSave = async () => {
+    if (!queries[0].data?.document) {
+      return;
+    }
+
+    setIsSaving(true);
+    try {
+      await updateDocumentContent({
+        documentId: queries[0].data.document.id,
+        organizationId: params.organizationId,
+        content: editedContent(),
+      });
+
+      await queries[0].refetch();
+      setIsEditing(false);
+      createToast({ type: 'success', message: 'Document content updated' });
+    } catch (_) {
+      createToast({ type: 'error', message: 'Failed to update document content' });
+    } finally {
+      setIsSaving(false);
+    }
```

Comment:
Prefer using the `safely` function (from my package `@corentinth/chisels` :innocent: ) instead of the try-catch-finally to avoid the indentation hell and scope isolation